iceoryx_binding_c/include/iceoryx_binding_c/types.h

```diff
@@ -125,10 +75,8 @@ typedef struct iox_pub_storage_t_ iox_pub_storage_t;
 
```

Comment:
We need to base the C publisher/subscriber on `UntypedPublisher`/`UntypedSubscriber` in order to get rid of the last magic numbers

Reply:
This will take some time, but can we leave this open as a separate issue and fix it later.

This PR is a huge improvement and would enable FreeBSD again when we check the subscriber/publisher size with less or equal.
So I would be for merging this and work on this in 2.1.

Reply:
Unfortunately it breaks cross-compilation, e.g. for QNX. But I have an idea how to fix this. It will become a bit messy, though.